src/dbSta/src/dbNetwork.cc

```diff
@@ -2562,4 +2607,159 @@ dbNetworkObserver::~dbNetworkObserver()
   }
 }
 
+/*
+  Hierarchical network api connections
+ */
+
+dbModule* dbNetwork::getParentModule(dbNet* net)
+{
+  if (hasHierarchy()) {
+    // get sink driver instance and return its parent
+    int drivingITerm = net->getDrivingITerm();
+    if (drivingITerm != 0 && drivingITerm != -1) {
+      dbITerm* iterm = block_->getITerm(drivingITerm);
+      dbModNet* modnet = iterm->getModNet();
+      if (modnet != nullptr) {
+        return modnet->getParent();
+      }
+    }
+    // default to top module
+    return block_->getTopModule();
+  }
+  return nullptr;
+}
+
+void dbNetwork::getInstanceTree(dbModule* start_module,
```

Comment:
Maybe getParentInstances ?  From the name I thought this would get the tree rooted at the start_module.

Reply:
agreed, that reads much better.